models/login_source.go

```diff
@@ -291,22 +297,32 @@ func GetLoginSourceByID(id int64) (*LoginSource, error) {
 	return source, nil
 }
 
+// ResetNonDefaultLoginSources clean other default source flag
+func ResetNonDefaultLoginSources(id int64) error {
+	_, err := x.NotIn("id", []int64{id}).Cols("is_default").Update(&LoginSource{IsDefault: false})
+	return err
+}
+
 // UpdateLoginSource updates information of login source to database or local file.
 func UpdateLoginSource(source *LoginSource) error {
 	if source.LocalFile == nil {
 		_, err := x.Id(source.ID).AllCols().Update(source)
+		if err != nil && source.IsDefault {
```

Comment:
Mixed error checking with a flag checking does not smell good to me.

Reply:
I will expand the logic.